dali/pipeline/operator/builtin/external_source.h

```diff
@@ -220,6 +220,8 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
         device_id_(spec.GetArgument<int>("device_id")),
         dtype_(spec.GetArgument<DALIDataType>("dtype")),
         previous_dtype_(DALIDataType::DALI_NO_TYPE),
+        ndim_(spec.GetArgument<int>("ndim")),
```

Comment:
More elaborate logic will be required if default value is dropped, but I still think it's worth it.

Reply:
done